usermods/pwm_outputs/usermod_pwm_outputs.h

```diff
@@ -0,0 +1,169 @@
+#pragma once
+#include "wled.h"
+
+#ifndef USERMOD_PWM_OUTPUT_PINS
+  #define USERMOD_PWM_OUTPUT_PINS 3
+#endif
+
+
+class PwmOutput {
+  public:
+
+    PwmOutput() {
+      DEBUG_PRINTLN("pwm_output[-1]: setup disabled output");
+    }
+
+    PwmOutput(int8_t pin, uint32_t freq) : pin_(pin), freq_(freq) {
+      DEBUG_PRINTF("pwm_output[%d]: setup to freq %d\n", pin_, freq_);
+      if (pin_ < 0 || !pinManager.allocatePin(pin_, true, PinOwner::UM_PWM_OUTPUTS)) {
+        return;
+      }
+      
+      channel_ = pinManager.allocateLedc(1);
+      if (channel_ == 255) {
+        DEBUG_PRINTF("pwm_output[%d]: failed to quire ledc\n", pin_);
+        pinManager.deallocatePin(pin_, PinOwner::UM_PWM_OUTPUTS);
+        return;
+      }
+
+      ledcSetup(channel_, freq_, bit_depth_);
+      ledcAttachPin(pin_, channel_);
+      DEBUG_PRINTF("pwm_output[%d]: init successful\n", pin_);
+      enabled_ = true;
+    }
+
+    void close() {
+      DEBUG_PRINTF("pwm_output[%d]: close\n", pin_);
+      if (!enabled_)
+        return;
+      pinManager.deallocatePin(pin_, PinOwner::UM_PWM_OUTPUTS);
+      if (channel_ != 255)
+        pinManager.deallocateLedc(channel_, 1);
+      channel_ = 255;
+      enabled_ = false;
+      duty_ = -1.0f;
+    }
+  
+    void setDuty(const float duty) {
+      DEBUG_PRINTF("pwm_output[%d]: set duty %f\n", pin_, duty);
+      if (!enabled_) {
+        return;
+      }
+      duty_ = min(1.0f, max(0.0f, duty));
+      uint32_t value = static_cast<uint32_t>((1 << bit_depth_) * duty_);
+      ledcWrite(channel_, value);
+    }
+
+    bool isEnabled() const {
+      return enabled_;
+    }
+
+    int8_t getPin() const {
+      return pin_;
+    }
+
+    uint32_t getFreq() const {
+      return freq_;
+    }
+
+    float getDuty() const {
+      return duty_;
+    }
+
+  private:
+    int8_t pin_ {-1};
```

Comment:
Why not use initialiser in default constructor as you did in parametric construcor? It is cleaner.
Be also aware that parametric constructor doesn't initialise all properties after the change.

```c++
PwmOutput() : pin(-1), freq_(50), ... {}
```

Reply:
With new logic there is no need for constructor at all, defaults stayed at variable declaration line.